### ISSUE TITLE
GET v1/products/attribute-sets/sets/list inconsistent return result

### ISSUE THREAD

**Issue by manhao-chen** [magento/magento2#11022]
### Preconditions
currently i'm using the magento 2.1.9 clean fresh install

### Steps to reproduce
i call the REST api with the following
http://URL/magento-219/rest/v1/products/attribute-sets/sets/list

### Expected result
expect the full list of attribute-sets to be returned

### Actual result
```
{"message":"%fieldName is a required field.","parameters":{"fieldName":"searchCriteria"}}
```

### Steps to reproduce with search fieldname filled in
i call the REST api with the following
http://URL/magento-219/rest/v1/products/attribute-sets/sets/list?searchCriteria[filter_groups][0][filters][0][field]=attribute_set_name&searchCriteria[filter_groups][0][filters][0][condition_type]=eq&searchCriteria[filter_groups][0][filters][0][value]=someAttributeSet

### Expected result
expect only someAttributeSet data returned

### Actual result
i get the full list of data, it completely **disregarded** my search query
```
{"items":[{"attribute_set_id":9,"attribute_set_name":"default2","sort_order":0,"entity_type_id":4},{"attribute_set_id":4,"attribute_set_name":"Default","sort_order":1,"entity_type_id":4}],"search_criteria":{"filter_groups":[{"filters":[{"field":"entity_type_code","value":"catalog_product","condition_type":"eq"}]}]},"total_count":2}
```

**Comment by RomanKrut** [comment]
@manhao-chen Hello. 
With request like http://URL/magento-219/rest/v1/products/attribute-sets/sets/list you must have some Search Criteria queries to get list of attribute sets, If you want to get only 1 attribute set from GET request use
/V1/products/attribute-sets/{attributeSetId}.
You can get more information here [Magento swagger](http://devdocs.magento.com/swagger/)

**Comment by manhao-chen** [comment]
I did provide some search criteria, but it appears to completely ignore them and return the full set of attributes-sets

**Comment by RomanKrut** [comment]
Yes i think so too , i just try to search like 
`?searchCriteria[filter_groups][0][filters][0][field]=attribute_set_name&searchCriteria[filter_groups][0][filters][0][value]=123`
but my response filter is always "filters":[
{
"field": "entity_type_code",
"value": "catalog_product",
"condition_type": "eq"
}

**Comment by magento-engcom-team** [comment]
@manhao-chen, thank you for your report.
We've created internal ticket(s) MAGETWO-80065 to track progress on the issue.

**Comment by TomashKhamlai** [comment]
These internal tickets [MAGETWO-73867](https://jira.corp.magento.com/browse/MAGETWO-73867), [MAGETWO-59723](https://jira.corp.magento.com/browse/MAGETWO-59723) exist for this issue.

**Comment by magento-team** [comment]
Internal ticket to track issue progress: MAGETWO-81904

**Comment by okorshenko** [comment]
The issue has been fixed in 2.2-develop branch and will be available with 2.2.2 release soon

**Comment by magento-team** [comment]
Internal ticket to track issue progress: MAGETWO-81910
